src/store/store.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { useEffect } from 'react'
+import { useNavigate } from 'react-router'
+import create, { SetState, State, StateCreator } from 'zustand'
+import createContext from 'zustand/context'
+import { persist } from 'zustand/middleware'
+
+import { SnackbarState, useSnackbar } from './snackbars'
+import { UploadStoreState, useUploadsManagerStore } from './uploads'
+
+const ZUSTAND_LOCAL_STORAGE_KEY = 'uploads_zustand'
+
+const { useStore, Provider } = createContext<RootState>()
+
+export type RootState = SnackbarState & UploadStoreState
+
+export const combineStateCreators = <TState extends State>(
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  ...stateCreators: StateCreator<any>[]
+): StateCreator<TState> => (set, get, api) => {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  let values: any = {}
+
+  stateCreators.forEach((sc) => {
+    values = Object.assign({}, values, sc(set, get, api))
+  })
+  return values
+}
+
+export const store = create<RootState>(
+  persist(combineStateCreators(useSnackbar, useUploadsManagerStore), {
+    name: ZUSTAND_LOCAL_STORAGE_KEY,
+    whitelist: ['uploadsState'],
+  })
+)
```

Comment:
So the state is currently spread, but we should be able to nest it, right? Current implementation force us to call store for each field separately:
```
  const uploadsProgress = useStore((state) => state.uploadsProgress)
  const updateAsset = useStore((state) => state.updateAsset)
  const uploadsState = useStore((state) => state.uploadsState)
```
Ideal api would be something similar to this:
```
  const { uploadsProgress, updateAsset, uploadsState}  = useStore((state) => state.uploads)
```